### PR TITLE
change: [M3-8058] - Add tags drawer for Kubernetes Cluster tag overflow

### PR DESCRIPTION
**Pull request by hkhalil-akamai**
## Description 📝
Changes the tags list in `KubeSummaryPanel` from an expanding list to a single line with an overflow drawer.

This PR also includes a refactor to `TagCell` that bundles the `TagDrawer` instead of requiring other components to include their own.

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Tag list takes up many lines when multiple tags are added Screenshot 2024-07-08 at 6 05 44 PM | Displays "..." button when overflowing Screenshot 2024-07-08 at 6 05 20 PM Tags drawer displays all tags  Screenshot 2024-07-08 at 6 06 02 PM |

### Verification steps
- Navigate to the details page of a Kubernetes cluster
- Add many tags and observe that instead of wrapping into multiple lines, the overflow is hidden and a button appears to display the tag drawer.

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/.changeset/pr-10658-changed-1720476630661.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Changed
+---
+
+Add drawer for overflowing tags in Kubernetes cluster details page ([#10658](https://github.com/linode/manager/pull/10658))
```
Comment (dwiley-akamai) [inline packages/manager/.changeset/pr-10658-changed-1720476630661.md:5]:
```suggestion
Display overflowing Kubernetes cluster tags in drawer ([#10658](https://github.com/linode/manager/pull/10658))
```

File: packages/manager/src/components/TagCell/AddTag.tsx [hunk 2]
```diff
@@ -63,6 +63,13 @@ export const AddTag = (props: AddTagProps) => {
 
   return (
     <Autocomplete
+      noOptionsText={
+        inputValue.length == 0 ? (
```
Comment (dwiley-akamai) [inline packages/manager/src/components/TagCell/AddTag.tsx:67]: Any particular reason to use `==` over `===` here?

**Review by dwiley-akamai — APPROVED** [review]
Overflowing tags (& non-overflowing tags) and drawer functioning as expected ✅
Code review ✅

<details>
<summary>This is an existing issue, but that panel is pretty rough when it comes to responsive design and we can get to an odd state where tags cover up the "Reset" button</summary>


Screenshot 2024-07-09 at 12 49 35 PM

</details>

File: packages/manager/src/components/TagCell/AddTag.tsx (hunk 2 above)
Comment (hkhalil-akamai) [reply packages/manager/src/components/TagCell/AddTag.tsx:67]: No, I'll switch to `===` just to be safe

**Comment by hkhalil-akamai** [comment]
> This is an existing issue, but that panel is pretty rough when it comes to responsive design and we can get to an odd state where tags cover up the "Reset" button

Yeah, it seems like that whole panel looks pretty inconsistent in mobile viewports. We can handle this in a follow-up ticket.

File: packages/manager/src/components/TagCell/TagCell.tsx [hunk 3]
```diff
@@ -41,6 +41,12 @@ export interface TagCellProps {
    * the tag list (i.e., by adding or deleting a tag).
    */
   updateTags: (tags: string[]) => Promise<any>;
+
+  /**
+   * Determines whether to allow tags to wrap or
+   * to overflow into a drawer.
+   */
+  view: 'inline' | 'panel';
```
Comment (hkhalil-akamai) [inline packages/manager/src/components/TagCell/TagCell.tsx:49]: Example of `'inline'` vs `'panel'`:


| Inline | Panel |
|-|-|
|Screenshot 2024-07-10 at 4 23 26 PM|Screenshot 2024-07-10 at 4 23 41 PM|

File: packages/manager/src/components/TagCell/TagCell.tsx [hunk 4]
```diff
@@ -59,29 +65,28 @@ const checkOverflow = (el: HTMLElement) => {
...
+  const [drawerOpen, setDrawerOpen] = React.useState(false);
 
   const [elRef, setElRef] = React.useState<HTMLDivElement | null>(null);
 
   const windowDimensions = useWindowDimensions();
 
   const [hasOverflow, setHasOverflow] = React.useState(false);
   React.useLayoutEffect(() => {
-    setHasOverflow(!!elRef && checkOverflow(elRef));
+    setHasOverflow(!!elRef && tags.length > 0 && checkOverflow(elRef));
```
Comment (hkhalil-akamai) [inline packages/manager/src/components/TagCell/TagCell.tsx:80]: This fixes a bug causing the ellipses button to display when there are no tags added.

File: packages/manager/src/components/TagCell/TagCell.tsx [hunk 5]
```diff
@@ -157,20 +162,27 @@ export const TagCell = (props: TagCellProps) => {
...
+          {view == 'inline' && <AddButton />}
         </StyledGrid>
       )}
+      {view == 'inline' && (
+        <TagDrawer
+          {...props}
+          onClose={() => setDrawerOpen(false)}
+          open={drawerOpen}
+        />
+      )}
```
Comment (hkhalil-akamai) [inline packages/manager/src/components/TagCell/TagCell.tsx:185]: Bundling the `TagDrawer` with `TagCell` to reduce boilerplate in components that use `TagCell` in the `'inline'` view.

File: packages/manager/src/features/Linodes/LinodeEntityDetailFooter.tsx [hunk 6]
```diff
@@ -18,22 +18,6 @@ import {
...
-  isSummaryView?: boolean;
-  linode: Linode;
-  openTagDrawer: (tags: string[]) => void;
-  variant?: TypographyProps['variant'];
-}
-
-export interface Props extends LinodeEntityDetailProps {
-  handlers: LinodeHandlers;
-}
-
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Linodes/LinodeEntityDetailFooter.tsx:36]: These exports were unused.

**Review by hkhalil-akamai — COMMENTED** [review]
Self-review

**Review by mjac0bs — APPROVED** [review]
Thanks for the fixes! (And good context in the self review, I was just most of the way through mine already. 😅 )

Checked Linodes, Domains, and Kube - left a couple nits, but did not see regressions and confirmed the issues explained in the PR are fixed...

...though that Kube summary would definitely benefit from a follow up ticket for layout refactor. Unknowingly, I had my zoom at 90% and saw the following wrapping issue with the container. It's an edge case.

https://github.com/linode/manager/assets/114685994/69e5bca1-55c5-4ea0-ae8a-b61778fc7401

**Comment by hkhalil-akamai** [comment]
> ...though that Kube summary would definitely benefit from a follow up ticket for layout refactor. Unknowingly, I had my zoom at 90% and saw the following wrapping issue with the container. It's an edge case.

Created M3-8348 to handle this and other layout inconsistencies. (cc @dwiley-akamai)

**Pull request merged**